pkg/network/events/monitor.go

```diff
@@ -54,13 +75,43 @@ func UnregisterHandler(handler ProcessEventHandler) {
 
 type eventHandlerWrapper struct{}
 
-func (h *eventHandlerWrapper) HandleEvent(ev *model.Event) {
+func (h *eventHandlerWrapper) HandleEvent(ev interface{}) {
+	if ev == nil {
+		log.Errorf("Received nil event")
+		return
+	}
+
+	evProcess, ok := ev.(*Process)
+	if !ok {
+		log.Errorf("Event is not a process")
+		return
+	}
+
 	m := theMonitor.Load()
 	if m != nil {
-		m.(*eventMonitor).HandleEvent(ev)
+		m.(*eventMonitor).HandleEvent(evProcess)
 	}
 }
 
+func (h *eventHandlerWrapper) Copy(ev *model.Event) interface{} {
+	m := theMonitor.Load()
+	if m != nil {
+		ev.ResolveFields()
+
+		var envCopy []string
+		copy(envCopy, ev.ProcessContext.EnvsEntry.Values)
```

Comment:
I don't think the `copy` here is going to do anything since `len(envCopy) == 0`. You will need to allocate `envCopy` with `make([]string, len(ev.ProcessContext.EnvsEntry.Values))` first.

Reply:
right right, thank you!